File: Features/attachment-sendfile/contracts/src/test/java/net.corda.samples.sendfile/states/InvoiceStateTests.java
package net.corda.samples.sendfile.states;

import net.corda.core.identity.CordaX500Name;
import net.corda.core.identity.Party;
import net.corda.testing.core.TestIdentity;
import org.junit.Test;

import java.util.Arrays;

import static org.junit.Assert.assertEquals;

public class InvoiceStateTests {

    private final Party a = new TestIdentity(new CordaX500Name("Alice", "", "GB")).getParty();
    private final Party b = new TestIdentity(new CordaX500Name("Bob", "", "GB")).getParty();

    private final String STRINGID = "StringID that is unique";

    @Test
    public void constructorTest() {
        InvoiceState st = new InvoiceState(STRINGID, Arrays.asList(a, b));
        assertEquals(STRINGID, st.getInvoiceAttachmentID());
    }


}

File: kotlin-samples/Features/attachment-sendfile/contracts/src/test/kotlin/net.corda.samples.sendfile/states/InvoiceStateTests.kt
package net.corda.samples.sendfile.contracts

import net.corda.core.identity.CordaX500Name
import net.corda.samples.sendfile.states.InvoiceState
import net.corda.testing.core.TestIdentity
import org.junit.Assert
import org.junit.Test

class InvoiceStateTests {
    private val a = TestIdentity(CordaX500Name("Alice", "", "GB")).party
    private val b = TestIdentity(CordaX500Name("Bob", "", "GB")).party
    private val STRINGID = "StringID that is unique"

    @Test
    fun constructorTest() {
        val (invoiceAttachmentID) = InvoiceState(STRINGID, listOf(a, b))
        Assert.assertEquals(STRINGID, invoiceAttachmentID)
    }
}
